dataclients/kubernetes/clusterstate.go

```diff
@@ -55,6 +51,26 @@ func (state *clusterState) GetEndpointsByService(namespace, name, protocol strin
 		TargetPort: servicePort.TargetPort.String(),
 	}
 
+	if cached, ok := state.cachedEndpoints[epID]; ok {
+		return cached
+	}
+
+	ep, ok := state.endpoints[epID.ResourceID]
+	if !ok {
+		return nil
+	}
+
+	targets := ep.targetsByServicePort(protocol, servicePort)
+	sort.Strings(targets)
+	state.cachedEndpoints[epID] = targets
+	return targets
+}
+
+func (state *clusterState) GetEndpointsByName(namespace, name, protocol string) []string {
+	epID := endpointID{
+		ResourceID: newResourceID(namespace, name),
+		Protocol:   protocol,
```

Comment:
Why can we avoid initializing Port here?

Reply:
I don't see port here

Reply:
```
type endpointID struct {
	definitions.ResourceID
	TargetPort string
	Protocol   string
}
```
I see TargetPort here. However, I don't see it initialized here.

Reply:
This func will do Endpoint lookup by name and namespace which identifies the kubernetes object.
TargetPort makes no sense in case we want all ports and not one specific port. For tests we use random ports on 127.0.0.1 so we can only listen on multiple ports, so targetport makes no sense in this case.